tracing/span.go

```diff
@@ -334,3 +343,104 @@ func (s *Span) preStop(err error) {
 		s.trace.setTag(ZipkinBinaryAnnotationKeyDebug, true)
 	}
 }
+
+// ForeachBaggageItem implements opentracing.SpanContext.
+//
+// We don't support any extra baggage items, so it's a noop.
+func (s *Span) ForeachBaggageItem(handler func(k, v string) bool) {}
+
+// SetBaggageItem implements opentracing.Span.
+//
+// As we don't support any extra baggage items,
+// it's a noop and just returns self.
+func (s *Span) SetBaggageItem(restrictedKey, value string) opentracing.Span {
+	return s
+}
+
+// BaggageItem implements opentracing.Span.
+//
+// As we don't support any extra baggage items, it always returns empty string.
+func (s *Span) BaggageItem(restrictedKey string) string {
+	return ""
+}
+
+// Finish implements opentracing.Span.
+//
+// It calls Stop with background context and nil error.
+func (s *Span) Finish() {
+	s.Stop(context.Background(), nil)
+}
+
+// FinishWithOptions implements opentracing.Span.
+//
+// In this implementation we ignore all timestamps in opts,
+// only extract context and error out of all the log fields,
+// and ignore all other log fields.
+//
+// Please use FinishOptions.Convert() to prepare the opts arg.
+func (s *Span) FinishWithOptions(opts opentracing.FinishOptions) {
+	var err error
+	ctx := context.Background()
+	for _, records := range opts.LogRecords {
+		for _, field := range records.Fields {
+			switch field.Key() {
+			case ctxKey:
+				if c, ok := field.Value().(context.Context); ok {
+					ctx = c
+				}
+			case errorKey:
+				if e, ok := field.Value().(error); ok {
+					err = e
+				}
+			}
+		}
+	}
+	s.Stop(ctx, err)
+}
+
+// Context implements opentracing.Span.
+//
+// It returns self as opentracing.SpanContext.
+func (s *Span) Context() opentracing.SpanContext {
+	return s
+}
+
+// SetOperationName implements opentracing.Span.
+func (s *Span) SetOperationName(operationName string) opentracing.Span {
+	s.trace.name = operationName
+	return s
+}
+
+// LogFields implements opentracing.Span.
+//
+// In this implementation it's a no-op.
+func (s *Span) LogFields(fields ...log.Field) {
+	// TODO
+}
+
+// LogKV implements opentracing.Span.
+//
+// In this implementation it's a no-op.
+func (s *Span) LogKV(alternatingKeyValues ...interface{}) {
+	// TODO
```

Comment:
Should this log something? How do we keep track of those TODOs?

Reply:
I'm not 100% sure how those APIs are used in real world yet. In the cases we care we likely would never need to use these 2 APIs, but leaving TODO here makes it more obvious that these are things that can be improved in the future when someone find some extra time :)

Reply:
Removed the TODOs.